public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -554,4 +545,16 @@ export function lokiSpecialRegexEscape(value: any) {
   return value;
 }
 
+/**
+ * Checks if the query expression uses function and so should return a time series instead of logs.
+ * Sometimes important to know that before we actually do the query.
+ */
+function isMetricsQuery(query: string): boolean {
```

Comment:
@ivanahuckova I think we can use this to check if we should run the instant query right? Cause right now we run the instant query also for logs query which I assume does not make much sense.

Reply:
Yes! That makes sense to use it also there.

Reply:
Cool will add a PR after this is merged.

Reply:
👍 awesome